File: desktopclient/src/main/scala/com/goexp/galgame/gui/db/mongo/BrandDB.java
package com.goexp.galgame.gui.db.mongo;

import com.goexp.common.db.mongo.DBOperatorTemplate;
import com.goexp.galgame.gui.model.Brand;

import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Updates.set;

public class BrandDB {

    private static DBOperatorTemplate tlp = new DBOperatorTemplate("galgame", "brand");

    public void updateWebsite(Brand item) {

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.updateOne(
                    eq(item.id)
                    , set("website", item.website)
            );
        });
    }


    public void updateIsLike(Brand item) {

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.updateOne(
                    eq(item.id)
                    , set("type", item.isLike.getValue())
            );
        });
    }

}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/db/mongo/BrandDB.scala
package com.goexp.galgame.gui.db.mongo

import com.goexp.db.mongo.DBOperator
import com.goexp.galgame.common.Config
import com.goexp.galgame.common.Config.DB_NAME
import com.goexp.galgame.gui.model.Brand
import com.mongodb.client.model.Filters
import com.mongodb.client.model.Updates.set


object BrandDB {

  private val tlp = new DBOperator(Config.DB_STRING, DB_NAME, "brand")

  def updateIsLike(item: Brand) =
    tlp.exec { documentMongoCollection =>
      documentMongoCollection.updateOne(Filters.eq(item.id), set("type", item.state.get().value))
    }
}